src/app/newSettings/Sections/Workspace/Billing/BillingWorkspaceSection.tsx

```diff
@@ -82,8 +82,10 @@ const BillingWorkspaceSection = ({ onClosePreferences }: BillingWorkspaceSection
   const onSaveBillingDetails = (newBillingDetails: CustomerBillingInfo) => {
     if (workspaceId) {
       setIsSavingBillingDetails(true);
-      dispatch(workspaceThunks.editWorkspace({ workspaceId, details: newBillingDetails })).then(() => {
-        setBillingDetails(newBillingDetails);
+      dispatch(workspaceThunks.editWorkspace({ workspaceId, details: newBillingDetails })).then((data) => {
+        if (data.payload) {
```

Comment:
add the payload in a more descriptive variable to know what is the data.
For example: `const isUpdateSuccessful = data.payload;`